reference-implementation/lib/writable-stream.js

```diff
@@ -359,6 +370,24 @@ function WritableStreamDefaultWriterClose(writer) {
   return promise;
 }
 
+
+function WritableStreamDefaultWriterCloseWithErrorPropagation(writer) {
+  const stream = writer._ownerWritableStream;
+
+  assert(stream !== undefined);
+
+  const state = stream._state;
+  if (state === 'closed' || state === 'closing') {
+    return Promise.resolve();
```

Comment:
Shouldn't this case handled as an error as it's already closed when we tried to propagate the close signal?